Authentication.md

```diff
@@ -153,29 +153,36 @@ And at last, you need to save the `location` header, which is the url that the u
 
 This will return another cookie that you need to save: `JSESSIONID`. To make sure you can save the cookie I recommend to disallow the HTTP request to follow redirects.
 
-### Step 2: Telling SOMToday who you are: `POST https://inloggen.somtoday.nl/?0-1.-panel-signInForm`
 
-#### Parameters
+### Step 2: Deciding if it is a username+password flow or an username first-flow
+
+When logging in somtoday, there'll be 2 options on how to send your username and password.
+
+username+password flow: After entering your school and pressing submit, there'll appear 2 input fields for username & password.
 
+username first-flow: After entering your school and pressing submit, there'll appear only 1 input field for the username.
+
+To decide which flow it is, we'll have to send one request.
+
+#### `POST /0-1.-panel-signInForm`
 | Name                                                     | Type      | Value                                 |
 |----------------------------------------------------------|-----------|---------------------------------------|
-| auth                                                     | Parameter | `authorization_code`                  |
-| loginLink                                                | body      | x                                     |
 | usernameFieldPanel:usernameFieldPanel_body:usernameField | body      | [username]                            |
-| origin                                                   | header    | https://inloggen.somtoday.nl          |
+| Origin                                                   | header    | https://inloggen.somtoday.nl          |
 | JSESSIONID                                               | cookie    | [JSESSIONID]                          |
+| auth                                                     | param     | `authorization_code`                  |
 | production-authenticator-stickiness                      | cookie    | [production-authenticator-stickiness] |
 
-`auth`: This is the authorization code that you got from the previous step.
+
+`auth`: This is the authorization code that you got from step 1.
 
 `username`: This is the username of the user that you want to authenticate.
 
 #### Returns
-This wil not return anything useful, you are sending your username to SOMToday, who will link you username to the `authorization_code` that you sent with the request.
 
+Don't follow the redirect, check if the ``auth`` parameter exists in the 'Location' header. If exists, then it is a **username + password flow**, otherwise it is an **username first-flow**
```

Comment:
```suggestion
Don't follow the redirect, check if the ``auth`` parameter exists in the 'Location' header. If it exists, then you need to use the **username + password flow**, otherwise it is a **username first-flow**
```